File: Golang/two-sum.go
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        tmp_dict = dict()
        for i, num in enumerate(nums):
            if num in tmp_dict:
                return [tmp_dict[num], i]
            else:
                tmp_dict[target - num] = i

File: Rust/two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut tmp_map = HashMap::new();
        let mut i = 0;
        for num in nums.iter() {
            match tmp_map.get(&(target - num)) {
                Some(&chord) => return vec![chord, i],
                None => {tmp_map.insert(num, i);}
            };
            i += 1;
        }
        vec![]
    }
}
